slash-commands/slash-commands.js

```diff
@@ -156,13 +199,123 @@ CandyShop.SlashCommands = (function(self, Candy, $) {
+			return false;
+		}
+		args = args.split(' ');
+
+		var user = new RegExp("^" + args[0] + "$", "i");
+		var userJid = null;
+		var room = new RegExp("^" + args[1] + "$", "i");
+		var roomJid = null;
+		var password = args[2];
+		
+		// loop through all rooms with current connections
+		$.each(Candy.Core.getRooms(), function(roomName, roomData) {
+			if( !roomJid && roomData && roomData.room.name.match(room) ) {
+				roomJid = roomName;
+				}
+			if( !roomJid && roomName.match(room) ) {
+				roomJid = roomName;
+				}
+
+			// loop through all users in a room
+			// compare jids, nicks and previous nicks
+			$.each(roomData.roster.getAll(), function(userName, userData) { 
+				if( !userJid && userData.getJid().match(user) ) {
+					userJid = userData.data.jid;
+					}
+				if( !userJid && userData.getNick().match(user) ){
+					userJid = userData.data.jid;
+					}
+				if( !userJid && userData.getPreviousNick() !== undefined && userData.getPreviousNick().match(user) ) {
+					userJid = userData.data.jid;
+					}
+				});
+			});
+
+
+		if(userJid !== undefined && userJid !== null && userJid !== '') {
+			if(room !== undefined && room !== '') {
+				if(password !== undefined && password !== '') {
+					Candy.View.Pane.Chat.onInfoMessage(self.currentRoom(), '', "Invited " + userJid + " to " + roomJid + " (with password)");
+					var stanza = $msg({'from': Candy.Core.getUser().data.jid, 'to': userJid, 'xmlns': 'jabber:client'}).c('x', {'xmlns': 'jabber:x:conference', 'jid': roomJid});
+					stanza.c("password").t(password);
+					Candy.Core.getConnection().send(stanza.tree());
+				}
+				else {
+					Candy.View.Pane.Chat.onInfoMessage(self.currentRoom(), '', "Invited " + userJid + " to " + roomJid);
+					var stanza = $msg({'from': Candy.Core.getUser().data.jid, 'to': userJid, 'xmlns': 'jabber:client'}).c('x', {'xmlns': 'jabber:x:conference', 'jid': roomJid});
+					Candy.Core.getConnection().send(stanza.tree());
+				}
+			}
+			else {
+				Candy.View.Pane.Chat.onInfoMessage(self.currentRoom(), '', "Invited " + userJid + " to " + self.currentRoom());
+				var stanza = $msg({'from': Candy.Core.getUser().data.jid, 'to': userJid, 'xmlns': 'jabber:client'}).c('x', {'xmlns': 'jabber:x:conference', 'jid': self.currentRoom()});
+				Candy.Core.getConnection().send(stanza.tree());
+			}
+		}
+		else {
+			Candy.View.Pane.Chat.onInfoMessage(self.currentRoom(), '', "Could not find " + args[0] + " to invite");
+		}
+
+	};
+
+	/** Function: kick
+	 * Kick user from current room. Admins only. 
+	 *
+	 * Parameters:
+	 *    (String) user Nickname of user, or JID - all currently connected rooms will be checked
+	 *    (String) comment Optional comment as to why they were kicked
+	 */
+	self.kick = function(args) {
+		if(args === undefined || args == ''){
+			Candy.View.Pane.Chat.onInfoMessage(self.currentRoom(), '', "usage: /kick nickname [comment] ");
+			return false;
+		}
+		args = args.split(' ');
```

Comment:
This won’t work with users whose nickname contains a space, you should introduce a way to escape those, the common practice in such UIs seems to be to use "" quotes around the nick.